tests/test_ops.py

```diff
@@ -21,155 +21,195 @@
-def test_displacement_heisenberg(phi, mag):
-    """ops: Tests the Heisenberg representation of the Displacement gate."""
-    r = mag
-    hbar = 2
-    matrix = cv.Displacement._heisenberg_rep([r, phi])
-    true_matrix = np.array(
-        [
-            [1, 0, 0],
-            [np.sqrt(2 * hbar) * r * np.cos(phi), 1, 0],
-            [np.sqrt(2 * hbar) * r * np.sin(phi), 0, 1],
-        ]
-    )
-    assert np.allclose(matrix, true_matrix)
-
-
-@pytest.mark.parametrize("phi", phis)
-@pytest.mark.parametrize("theta", phis)
-def test_beamsplitter_heisenberg(phi, theta):
-    """ops: Tests the Heisenberg representation of the Beamsplitter gate."""
-    matrix = cv.Beamsplitter._heisenberg_rep([theta, phi])
-    true_matrix = np.array(
-        [
-            [1, 0, 0, 0, 0],
+class TestQubit:
+    """Tests the qubit based operations."""
+
+    identity = np.array([[1, 0], [0, 1]])
+    paulix = np.array([[0, 1], [1, 0]])
+    pauliy = np.array([[0, -1j], [1j, 0]])
+    pauliz = np.array([[1, 0], [0, -1]])
+    hadamard = 1/np.sqrt(2)*np.array([[1, 1],[1, -1]])
+
+    observables = [identity, paulix, pauliy, pauliz, hadamard]
+
+
+    @pytest.mark.parametrize("observable, eigvals, eigvecs", [
+        (np.array([[1, 0], [0, 1]]), np.array([1., 1.]), np.array([[1., 0.],[0., 1.]])),
+        (np.array([[0, 1], [1, 0]]), np.array([-1, 1]), np.array([[-0.70710678,  0.70710678],[ 0.70710678,  0.70710678]])),
+        (np.array([[0, -1j], [1j, 0]]), np.array([-1, 1]), np.array([[-0.70710678+0.j        , -0.70710678+0.j        ], [ 0.        +0.70710678j,  0.        -0.70710678j]])),
+        (np.array([[1, 0], [0, -1]]), np.array([-1, 1]), np.array([[0., 1.], [1., 0.]])),
+        (1/np.sqrt(2)*np.array([[1, 1],[1, -1]]), np.array([-1, 1]), np.array([[ 0.38268343, -0.92387953],[-0.92387953, -0.38268343]])),
+    ])
+    def test_hermitian_eigvals_eigvecs(self, observable, eigvals, eigvecs, tol):
+        """Tests that the eigvals method of the Hermitian class returns the correct results."""
+        key = tuple(observable.flatten().tolist())
+
+        qubit.Hermitian.eigvals(observable)
+        assert np.allclose(qubit.Hermitian._eigs[key]["eigval"], eigvals, atol=tol, rtol=0)
+        assert np.allclose(qubit.Hermitian._eigs[key]["eigvec"], eigvecs, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("observable, eigvals, eigvecs", [
+        (np.array([[1, 0], [0, 1]]), np.array([1., 1.]), np.array([[1., 0.],[0., 1.]])),
+        (np.array([[0, 1], [1, 0]]), np.array([-1, 1]), np.array([[-0.70710678,  0.70710678],[ 0.70710678,  0.70710678]])),
+        (np.array([[0, -1j], [1j, 0]]), np.array([-1, 1]), np.array([[-0.70710678+0.j        , -0.70710678+0.j        ], [ 0.        +0.70710678j,  0.        -0.70710678j]])),
+        (np.array([[1, 0], [0, -1]]), np.array([-1, 1]), np.array([[0., 1.], [1., 0.]])),
+        (1/np.sqrt(2)*np.array([[1, 1],[1, -1]]), np.array([-1, 1]), np.array([[ 0.38268343, -0.92387953],[-0.92387953, -0.38268343]])),
+    ])
```

Comment:
Since this test data is identical to the above, I'm in favour of moving it to a module variable (i.e., `EIGVALS_TEST_DATA`) at the top of the test file, and then just calling it:
```python
    @pytest.mark.parametrize("observable, eigvals, eigvecs", EIGVALS_TEST_DATA)
```

I find this makes the tests easier to read even when test data isn't repeated, and would prefer it to be the case across the board. Ideally, the eyes should focus on the tests themselves rather than the test data.